operators/src/adapters/raster_subquery_adapter.rs

```diff
@@ -198,64 +204,79 @@ where
-            Some(fut) => {
-                ready!(fut.poll(cx))
-            }
-            None => {
-                debug!("running_fold is None");
-                return Poll::Ready(None); // should initialize next tile query?
+            let query_future_result =
+                if let Some(query_future) = this.running_query.as_mut().as_pin_mut() {
+                    // TODO: match block?
+                    let query_result: Result<BoxStream<'a, Result<RasterTile2D<PixelType>>>> =
+                        ready!(query_future.poll(cx));
+                    Some(query_result)
+                } else {
+                    None
+                };
+
+            this.running_query.set(None);
+
+            match query_future_result {
+                Some(Ok(tile_query_stream)) => {
+                    let tile_folding_accu = this
+                        .sub_query
+                        .new_fold_accu(fold_tile_spec, tile_query_rectangle)?;
+
+                    debug!("New running_fold: {:?}", &tile_query_rectangle);
+                    let tile_folding_stream =
+                        tile_query_stream.try_fold(tile_folding_accu, this.sub_query.fold_method());
+
+                    this.running_fold.set(Some(tile_folding_stream));
+                }
+                Some(Err(err)) => {
+                    debug!("Tile fold stream returned error: {:?}", &err);
+                    *this.ended = true;
+                    return Poll::Ready(Some(Err(err)));
+                }
+                None => {} // there is no result but there might be a running fold...
             }
-        };
 
-        // set the running future to None --> will create a new one in the next call
-        this.running_fold.set(None);
+            let future_result = match this.running_fold.as_mut().as_pin_mut() {
+                Some(fut) => {
+                    ready!(fut.poll(cx))
+                }
+                None => {
+                    debug!("running_fold is None");
+                    return Poll::Ready(None); // should initialize next tile query?
+                }
+            };
 
-        let tile_accu_result = future_result.map(FoldTileAccu::into_tile);
+            // set the running future to None --> will create a new one in the next call
+            this.running_fold.set(None);
+
+            let tile_accu_result: Result<RasterTile2D<PixelType>> =
+                future_result.map(FoldTileAccu::into_tile);
+
+            tile_accu_result
+        } else {
+            // no sub query query rectangle was produced for the current tile, thus output an empty tile
+            Ok(RasterTile2D::new_with_tile_info(
+                this.query_rect.time_interval,
+                fold_tile_spec,
+                GridOrEmpty::Empty(EmptyGrid2D::<PixelType>::new(
+                    fold_tile_spec.tile_size_in_pixels,
+                    this.no_data_value.unwrap_or(PixelType::zero()),
```

Comment:
maybe add a TODO that this is a fallback assumption

Reply:
689bfd60e5d22148aac2b05d4f5b6a8e7141092c